### PR TITLE
Adding/moving in tests from other repos + improving CI prep

### PR DESCRIPTION
**Pull request by oybed**
#### What does this PR do?
Structuring the `test` area of `openshift-applier` with additional tests sourced from other repos (e.g.: casl-ansible, cluster-lifecycle, etc.). This is preparation for enabling CI for the repo. 

#### How should this be tested?
See `test/README.md`

#### Is there a relevant Issue open for this?
N/A

#### Who would you like to review this?
cc: @redhat-cop/openshift-applier

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+> ansible-playbook playbooks/openshift-cluster-seed.yml -i <path-to-inventory>
+
+List of example test runs:
+
+```
+ansible-playbook playbooks/openshift-cluster-seed.yml -i test/inventories/multi-files-dir
+ansible-playbook playbooks/openshift-cluster-seed.yml -i test/inventories/multi-params-dir
+ansible-playbook playbooks/openshift-cluster-seed.yml -i test/inventories/params-from-file
+ansible-playbook playbooks/openshift-cluster-seed.yml -i test/inventories/params-from-vars
+ansible-playbook playbooks/openshift-cluster-seed.yml -i test/inventories/pre-post-steps
```
Comment (etsauer) [inline test/README.md]: This test fails to execute with:
```
TASK [openshift-applier : Run ansible-galaxy to pull in dependency roles] ****************************************************************************************************************************************************************************************************************
failed: [localhost] (item=/home/esauer/src/openshift-applier/test/inventories/pre-post-steps/../../files/galaxy/requirements.yml) => {"changed": true, "cmd": ["ansible-galaxy", "install", "-r", "/home/esauer/src/openshift-applier/test/inventories/pre-post-steps/../../files/galaxy/requirements.yml", "-p", "/tmp/ansible.GglKOe/"], "delta": "0:00:00.424059", "end": "2018-08-09 15:00:18.452520", "item": "/home/esauer/src/openshift-applier/test/inventories/pre-post-steps/../../files/galaxy/requirements.yml", "msg": "non-zero return code", "rc": 1, "start": "2018-08-09 15:00:18.028461", "stderr": "ERROR! Unable to open /home/esauer/src/openshift-applier/test/inventories/pre-post-steps/../../files/galaxy/requirements.yml: [Errno 2] No such file or directory: u'/home/esauer/src/openshift-applier/test/inventories/pre-post-steps/../../files/galaxy/requirements.yml'", "stderr_lines": ["ERROR! Unable to open /home/esauer/src/openshift-applier/test/inventories/pre-post-steps/../../files/galaxy/requirements.yml: [Errno 2] No such file or directory: u'/home/esauer/src/openshift-applier/test/inventories/pre-post-steps/../../files/galaxy/requirements.yml'"], "stdout": "", "stdout_lines": []}
```

**Comment by oybed** [comment]
@etsauer thx for testing. Turns out there's an entry in .gitignore that caused that file to not get pushed. I have updated the location/inventory to fix this. Please re-pull and re-try.

File: test/README.md (hunk 1 above)
Comment (etsauer) [reply test/README.md]: Now I'm getting:
```TASK [/tmp/ansible.Es8tQE/casl-ansible/roles/openshift-labels : Apply label openshift-applier=true to object route] **********************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["oc", "label", "--overwrite", "route", "route1-test", "openshift-applier=true"], "delta": "0:00:00.500971", "end": "2018-08-09 16:06:55.153388", "msg": "non-zero return code", "rc": 1, "start": "2018-08-09 16:06:54.652417", "stderr": "Error from server (NotFound): routes.route.openshift.io \"route1-test\" not found", "stderr_lines": ["Error from server (NotFound): routes.route.openshift.io \"route1-test\" not found"], "stdout": "", "stdout_lines": []}

RUNNING HANDLER [openshift-applier : Clean up temporary dependency dir] ******************************************************************************************************************************************************************************************************************
	to retry, use: --limit @/home/esauer/src/openshift-applier/playbooks/openshift-cluster-seed.retry

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=34   changed=4    unreachable=0    failed=1   
```

Comment (oybed) [reply test/README.md]: @etsauer congratulation, that's an actual breakage in `openshift-applier`, so the test is valid, but we need to fix openshift-applier. CI is already paying off. :-)

Comment (oybed) [reply test/README.md]: @etsauer this bug has now been fixed (in the `master` branch) - do you mind re-testing?

**Comment by oybed** [comment]
@etsauer  bringing the comments out to the main conversation. The issue you hit when running the pre/post steps tests have now been fixed in the `openshift-applier` role (a true bug). Can you please re-try and merge if you are ok with the current version?

File: tests/files/secrets/project1.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: v1
+kind: List
+items:
+- kind: ProjectRequest
+  apiVersion: v1
+  metadata:
+    name: oa-ci-secret1
+  displayName: OpenShift Applier Secret Test 1 (displayName)
```
Comment (etsauer) [inline tests/files/secrets/project1.yml]: Not that it matters, but you've got some instances of `display_name`, and some of `displayName`

Comment (oybed) [reply tests/files/secrets/project1.yml]: Updated

**Pull request merged**